omnilib/prototypes/functions/recipe-standard.lua

```diff
@@ -341,45 +248,12 @@ function omni.marathon.standardise(recipe)
 	---------------------------------------------------------------------------
 	-- Icons standardisation
 	---------------------------------------------------------------------------
-	if recipe.icons and recipe.icon then -- case both, replace icons with icon (assume icon is new)
-		--replace recipe.icons with the new
-		recipe.icons[1]={icon=recipe.icon,icon_size=recipe.icon_size or 32,icon_mipmaps=recipe.icon_mipmaps or nil}
-	elseif recipe.icon then --only icon, set icons
-		recipe.icons={{icon=recipe.icon,icon_size=recipe.icon_size or 32,icon_mipmaps=recipe.icon_mipmaps or nil}}
-	 -- case neither icon or icons exist and recipe has ~= 1 results (search via product) (with 1 result its using the item icons)
-	elseif (not recipe.icons and not recipe.icon and ((recipe.normal.results and #recipe.normal.results ~= 1) or (recipe.expensive.results and #recipe.expensive.results ~= 1))) then
-		----------------------------------------------
-		-- NO RECIPE ICONS, SEARCH FOR MAIN PRODUCT --
-		----------------------------------------------
-		if (recipe.main_product and recipe.main_product ~= "") then
-			res=recipe.main_product
-		elseif (recipe.normal.main_product and recipe.normal.main_product ~= "") then
-			res=recipe.normal.main_product
-		----------------------------------------------
-		-- NO MAIN PRODUCT, SEARCH FOR FIRST RESULT --
-		----------------------------------------------
-		elseif recipe.result and recipe.result.name then
-			res=recipe.result.name
-		elseif recipe.result then
-			res=recipe.result
-		elseif recipe.normal.result and recipe.normal.result.name then
-			res=recipe.normal.result.name
-		elseif recipe.normal.result then
-			res=recipe.normal.result
-		elseif recipe.results and recipe.results[1] and recipe.results[1].name then
-			res=recipe.results[1].name
-		elseif 	recipe.normal.results and recipe.normal.results[1] and recipe.normal.results[1].name then
-			res=recipe.normal.results[1].name
-		else
-			res=recipe.name --should never get this far...
-		end
-		--find res
-		local item = omni.lib.find_prototype(res)
-		if item then
-			recipe.icons=set_icons_tab(item)
-		end
+	if recipe.icons or recipe.icon then -- case both, replace icons with icon (assume icon is new)
+		recipe.icons = omni.lib.find_result_icon(recipe)
+		-- nil out non-compliant
+		recipe.icon = nil
+		recipe.icon_size = nil
+		recipe.icon_mipmaps = nil
```

Comment:
Bad, we want to avoid using the icons of a result if recipe icon(s) exist.
Incase .icons AND ,icon exist, use icon and icon_size
if only .icon exists, move it into .icons
if only .icons exist everything is fine

Only if none of these is true, we want to look at main product/result item icons

Reply:
The base game precedence is .icons overrules .icon. Is there a reason we do the reverse?

Reply:
None that i can think of, that got brought over from all code.
But i agree with prefering .icons, the checks described above are still needed tho